.github/workflows/build-test.yml

```diff
@@ -105,17 +106,31 @@ jobs:
         uses: actions/checkout@v3
       - name: Install iceoryx dependencies and clang-tidy
         uses: ./.github/actions/install-iceoryx-deps-and-clang
-      - name: build with sanitizers
-        run: ./tools/ci/build-test-ubuntu-with-sanitizers.sh clang
+      - name: Run Address Sanitizer
+        run: ./tools/ci/build-test-ubuntu-with-sanitizers.sh clang asan
 
-  build-test-macos-with-sanitizers:
-    # prevent stuck jobs consuming runners for 6 hours
+  build-test-ubuntu-with-thread-sanitizer-clang-latest:
+    # prevent stuck jobs consuming runners for 3 hours
+    timeout-minutes: 180
+    runs-on: ubuntu-latest
+    needs: pre-flight-check
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install iceoryx dependencies and clang-tidy
+        uses: ./.github/actions/install-iceoryx-deps-and-clang
+      - name: Run Thread Sanitizer
+        run: ./tools/ci/build-test-ubuntu-with-sanitizers.sh clang tsan
```

Comment:
@elBoberido 

I can implement 5 :
- first job builds only
- the other three jobs run in parallel one of hoofs_moduletests, posh_moduletests or posh_integrationtests

As I checked, I need to use  `actions/upload-artifact` and  `actions/upload-artifact`  to share test binaries between build job and test job. 

cc @dkroenke

Reply:
@ibrhmkuru would be great but since this PR is already in a good state I think we should go for option 2 in this PR and do 5 in a follow up. This has the advantage that someone can look at the warnings and check for false positives.

 Btw. I did something like option 5 for iceoryx-rs
- https://github.com/eclipse-iceoryx/iceoryx-rs/tree/master/.github
- https://github.com/eclipse-iceoryx/iceoryx-rs/actions/runs/2728414816
Just in case you need some inspiration :)

cc @dkroenke

Reply:
@elBoberido Thanks for the example for inspiration :) I've done option 2 as suggested. It's ready for review.

cc @dkroenke